logic/strategy/card_utils.go

```diff
@@ -36,41 +36,28 @@ func otherOptions(desired int, avoid map[model.Card]struct{}) [][]model.Card {
 	return options
 }
 
-func chooseFrom(desired int, hand []model.Card) [][]model.Card { //nolint:gocyclo
-	if desired > len(hand) || desired > 4 || desired <= 0 {
-		return nil
-	}
-	// 6 choose 2 = 15, our largest number of potential deposits to the crib
-	allDeposits := make([][]model.Card, 0, 15)
-	for i, c1 := range hand {
-		if desired == 1 {
-			// it's a three or four player game, return one card
-			allDeposits = append(allDeposits, []model.Card{c1})
-			continue
+func chooseFrom(k int, hand []model.Card) [][]model.Card {
+	if k == 1 {
+		all := make([][]model.Card, len(hand))
+		for i, e := range hand {
+			all[i] = []model.Card{e}
 		}
-		for j := i + 1; j < len(hand); j++ {
-			c2 := hand[j]
-			if desired == 2 {
-				allDeposits = append(allDeposits, []model.Card{c1, c2})
-				continue
-			}
-			for k := i + 1; k < len(hand); k++ {
-				c3 := hand[k]
-				if desired == 3 {
-					allDeposits = append(allDeposits, []model.Card{c1, c2, c3})
-					continue
-				}
-				for l := i + 1; l < len(hand); l++ {
-					c4 := hand[l]
-					if desired == 4 {
-						allDeposits = append(allDeposits, []model.Card{c1, c2, c3, c4})
-						continue
-					}
-				}
-			}
+		return all
+	}
+	if k == len(hand) {
+		return [][]model.Card{hand}
+	}
+	all := make([][]model.Card, 0)
+	for i := 0; i <= len(hand)-k; i++ {
+		c := hand[i]
+		others := hand[i+1:]
+		otherSets := chooseFrom(k-1, others)
+		for _, s := range otherSets {
+			set := append([]model.Card{c}, s...)
```

Comment:
not sure what the best way to do this is, from either a memory or speed concern. Maybe we should just `make` new slices each time?

```go
set := make([]model.Card, 1, k)
set[0] = c
set = append(set, s...)
```

I don't know what the benchmarks would show, but I'd guess that doing an append like it is now may not be the most space efficient way. Not sure how golang would handle it 🤷‍♂

Reply:
Sure - I can do that. Which part of the current implementation concerns you?